### PR TITLE
PUBDEV-8874: eval_metric can be scored on validation set (internal xgb)

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-core/src/test/java/water/fvec/FrameTest.java [hunk 1]
```diff
@@ -508,4 +509,35 @@ public void testToCSVEscapedQuotes() throws Exception {
...
+  public void testMakeSimilarlyDistributed() {
+    try {
+      Scope.enter();
+      final int[] dataBig = ArrayUtils.seq(0, 1001);
+      final int[] dataSmall = ArrayUtils.seq(0, 231);
+
+      final Frame big = new TestFrameBuilder()
+              .withVecTypes(Vec.T_NUM)
+              .withDataForCol(0, dataBig)
+              .withChunkLayout(10L, 0L, 300L, 400L, 50L, 0, 0, 241, 0)
```
Comment (kanech) [inline h2o-core/src/test/java/water/fvec/FrameTest.java:522]: Using suffix for long (L) just for half of the arguments seems oddly.

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java [hunk 2]
```diff
@@ -774,11 +778,11 @@ private boolean doScoring(final XGBoostModel model, final XGBoostExecutor exec,
     }
   }
 
-  static CustomMetric toCustomMetric(EvalMetric evalMetric) {
+  static CustomMetric toCustomMetric(EvalMetric evalMetric, boolean isTrain) {
```
Comment (kanech) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java:781]: The boolean parameter looks like a code smell to me. A reader has to switch to implementation to see what's going on. IMHO, two methods like toCustomTrainingMetric() and toCustomValidationMetric() would be much more expressive.

**Review by kanech — APPROVED** [review]
LGTM

File: h2o-core/src/test/java/water/fvec/FrameTest.java (hunk 1 above)
Comment (michalkurka) [reply h2o-core/src/test/java/water/fvec/FrameTest.java:522]: ouch! thanks for noticing

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java (hunk 2 above)
Comment (michalkurka) [reply h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java:781]: sounds good

Comment (michalkurka) [reply h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java]: add, code looks better now - thx
